server/src/main/java/org/opensearch/indices/ShardLimitValidator.java

```diff
@@ -224,9 +255,12 @@ static Optional<String> checkShardLimit(int newShards, ClusterState state, int m
             return Optional.empty();
         }
         int maxShardsPerNode = maxShardsPerNodeSetting;
-        int maxShardsInCluster = maxShardsPerNode * nodeCount;
-        int currentOpenShards = state.getMetadata().getTotalOpenIndexShards();
+        int maxShardsInCluster = maxShardsPerClusterSetting;
```

Comment:
What happens if node level limit is 100, cluster level limit is 400 and we have 3 nodes.
If we try to increase total number of shards to 320, will it allow?

Reply:
As of now yes. Should we have an upper cap as well on the maximum number of shards that can grow on a cluster? I was thinking of not allowing above ```maxShardsPerNode * nodeCount``` value.

Reply:
IMO, we should check for `min(maxShardsPerClusterSetting, maxShardsPerNode * nodeCount)`. What do you think?

@gbbafna @Bukhtawar Need your input as well.

Reply:
Yes it should be min

Reply:
Addressed the comment.